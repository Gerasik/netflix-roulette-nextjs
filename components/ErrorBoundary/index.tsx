/* eslint-disable no-console */
import React, { ErrorInfo } from 'react';
import { ErrorBoundaryState } from './models';

export default class ErrorBoundary extends React.Component<{}, ErrorBoundaryState> {
  constructor(props) {
    super(props);
    this.state = { hasError: false };
  }

  static getDerivedStateFromError(): ErrorBoundaryState {
    return { hasError: true };
  }

  componentDidCatch(error: Error, errorInfo: ErrorInfo): void {
    console.log('---log error---');
    console.log('ErrorBoundaryState');
    // console.log(error, errorInfo);
    console.log('---log error---');
  }

  render() {
    const { children } = this.props;
    const { hasError } = this.state;
    if (hasError) {
      return <h1>Something went wrong.</h1>;
    }

    return children;
  }
}
