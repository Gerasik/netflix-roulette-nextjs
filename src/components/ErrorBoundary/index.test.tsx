/* eslint-disable @typescript-eslint/camelcase */
import React from 'react';
import { shallow, mount } from 'enzyme';

import ErrorBoundary from './index';

const Buggy: React.FunctionComponent<{ throwError?: boolean }> = ({ throwError = true }) => {
  if (throwError) {
    throw new Error('An error has occurred in Buggy component!');
  }

  return null;
};

describe('<ErrorBoundary /> test data', () => {
  const wrapper = shallow(
    <ErrorBoundary>
      <h1>Hello world</h1>
    </ErrorBoundary>
  );

  it('should render with test data', () => {
    expect(wrapper).toMatchSnapshot();
  });
});

describe('<ErrorBoundary /> render error', () => {
  const wrapper = mount(
    <ErrorBoundary>
      <Buggy />
    </ErrorBoundary>
  );

  it('should render with test data', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
