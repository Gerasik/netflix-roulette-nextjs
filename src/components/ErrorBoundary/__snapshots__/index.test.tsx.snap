// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorBoundary /> render error should render with test data 1`] = `
<ErrorBoundary>
  <h1>
    Something went wrong.
  </h1>
</ErrorBoundary>
`;

exports[`<ErrorBoundary /> test data should render with test data 1`] = `
<h1>
  Hello world
</h1>
`;
